test/testdata/lsp/hover_ampersand_operations.rb

```diff
@@ -39,4 +49,13 @@ def main
   # ^ hover: T.nilable(String)
                     # ^ hover: sig {returns(String)}
                      # ^ hover: sig {returns(String)}
+
+                    
+
+  # Examples of where the detection of dead branch loc is "weird" to me
+  str_breed = dog.breed&.to_s # here the unreachable is at dog, which is not close to the &
+  cat_breed = Animal::Cat.new&.breed # here the unreachable code is at Animal
+  maybe_dog_breed_forced = T.must(maybeDog)&.breed # here the unreachable code is at T
+  @dog = dog
+  own_dog_breed = @dog&.breeed # here the unreachable code is at @
```

Comment:
Since these aren't LSP-related they should probably live in their own test file in `test/testdata/desugar`. Having small test cases are nice, because they run more quickly and are easier to put under a debugger. And not putting them in LSP makes it clear that this feature is not impacted by LSP in any way.

Reply:
Yeah, these were not intended to be here at all.. I just pushed them so you had a reproducible case. If we end up deciding to merge this PR w/o the autocorrector, I'll just remove the latest commit and can report these bugs as a separate issue.